src/runtime/virtcontainers/kata_agent.go

```diff
@@ -178,6 +182,32 @@ func getSandboxPath(id string) string {
 	return filepath.Join(kataHostSharedDir(), id)
 }
 
+// Use in nydus case, guest shared dir is compatible with virtiofsd sharedir
+// nydus images are presented in kataGuestNydusImageDir
+//
+// virtiofs mountpoint: "/run/kata-containers/shared/"
+// kataGuestSharedDir: "/run/kata-containers/shared/containers"
+// kataGuestNydusImageDir: "/run/kata-containers/shared/images"
+var kataGuestNydusRootDir = func() string {
+	if rootless.IsRootless() {
+		// filepath.Join removes trailing slashes, but it is necessary for mounting
+		return filepath.Join(rootless.GetRootlessDir(), defaultKataGuestNydusRootDir) + "/"
+	}
+	return defaultKataGuestNydusRootDir
+}
+
+var imageMountPath = func(cid string) string {
+	return filepath.Join("/images", cid, lowerDir)
+}
+
+var kataGuestNydusImageDir = func() string {
```

Comment:
Easy to unit test.

Reply:
done, add unit test in kata_agent_test.go